core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -455,29 +517,55 @@ KOKKOS_INLINE_FUNCTION void parallel_for(
     const Impl::TeamVectorRangeBoundariesStruct<iType, Impl::SYCLTeamMember>&
         loop_boundaries,
     const Closure& closure) {
-  for (auto i = loop_boundaries.start; i != loop_boundaries.end; ++i)
+  // FIXME_SYCL adapt for vector_length != 1
```

Comment:
Why does this have a fixme? Isn't this already correct if you encode team_size*vector_length in the first dim?

Reply:
I think we want the first dimension to be `team_size` and the second one `vector_length` so we need to consider the linear (combined) index here.
I rather mark too many places to revisit later for `vector_length>1` than too few.